pages/activity/index.js

```diff
@@ -0,0 +1,77 @@
+import Head from "next/head";
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import Button from "@/components/ui/Button";
+import Header from "@/components/layout/Header";
+import ActivityList from "@/components/layout/ActivityList";
+import ActivityForm from "@/components/layout/ActivityForm";
+import ActivityFilter from "@/components/layout/ActivityFilter";
+
+export default function HomePage({
+  handleAddActivity,
+  activities,
+  bookmarks,
+  toggleBookmark,
+  handleFilter,
+  filter,
+}) {
+  const [showForm, setShowForm] = useState(false);
+  const [showFilter, setShowFilter] = useState(false);
+
+  const activity = {
```

Comment:
you hardcoded the activity inside the component. Consider to create a lib-folder and an activity.js because you will definitely use it more often